Bottom: 0006f544e9e9283a7fdfdad55e1032a2af7737b0
Top:    63c40cc335ef6a1240516b80965b950fd461eca1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-05 18:44:15 +0100

find.c: Refactor some common sequences into a helper function


---

diff --git a/src/find.c b/src/find.c
index 99893da..819fa44 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3321,6 +3321,15 @@ struct drc_info
   int flag;
 };
 
+static void
+start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
+                      int flag, bool AndDraw)
+{
+  ListStart (type, ptr1, ptr2, ptr3, flag);
+  DoIt (flag, true, AndDraw);
+  DumpList ();
+}
+
 /*-----------------------------------------------------------------------------
  * Check for DRC violations on a single net starting from the pad or pin
  * sees if the connectivity changes when everything is bloated, or shrunk
@@ -3338,10 +3347,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   if (PCB->Shrink != 0)
     {
       Bloat = -PCB->Shrink;
-      ListStart (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG);
-      DoIt (DRCFLAG | SELECTEDFLAG, true, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false);
       /* ok now the shrunk net has the SELECTEDFLAG set */
-      DumpList ();
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
       drc = true;               /* abort the search if we find anything not already found */
@@ -3353,14 +3360,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           User = true;
           drc = false;
           Bloat = -PCB->Shrink;
-          ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-          DoIt (SELECTEDFLAG, true, true);
-          DumpList ();
-          ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
           Bloat = 0;
           drc = true;
-          DoIt (FOUNDFLAG, true, true);
-          DumpList ();
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3393,9 +3396,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
   Bloat = 0;
-  ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-  DoIt (SELECTEDFLAG, true, false);
-  DumpList ();
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3408,14 +3409,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       User = true;
       drc = false;
       Bloat = 0;
-      ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-      DoIt (SELECTEDFLAG, true, true);
-      DumpList ();
-      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true);
       Bloat = PCB->Bloat;
       drc = true;
-      DoIt (FOUNDFLAG, true, true);
-      DumpList ();
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3443,9 +3440,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
       Bloat = 0;
-      ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag);
-      DoIt (flag, true, true);
-      DumpList ();
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
